## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :x: [more...](#e9532322-c686-4c74-a2a3-a55adad377b1) |
| mint-fs.sh | :x: [more...](#1ff432f6-63cc-4baa-bc30-ec10e375104e) |
| mint-gateway-s3.sh | :x: [more...](#50c6bebf-1a6c-4806-822f-eefe779452b9) |
| mint-erasure.sh | :x: [more...](#b643b42b-dba1-41d0-a71a-8de103719936) |
| mint-dist-erasure.sh | :x: [more...](#73adc3b3-b42d-4542-964b-f68f94e94955) |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#422f58bf-b33b-434d-93dc-1e5b73dee1a6) |
| mint-pools.sh | :x: [more...](#909a763d-accd-45a3-8233-d225a21dcc55) |
<a name="b643b42b-dba1-41d0-a71a-8de103719936"></a>
### 15511-dd6105f/mint-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s6:30275
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 7 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-3a2c5hncw5fgva",
    "objectName": "p1d3vkkyjxetfrpzsa0leg51w3o2zo"
  },
  "duration": 13,
  "function": "testObjectTagging",
  "message": "AWS SDK Go PUTObject Tag input did not match with GetObjectTagging output <nil>",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-java tests ... done in 1 seconds
(2/14) Running aws-sdk-php tests ... done in 43 seconds
(3/14) Running aws-sdk-ruby tests ... done in 4 seconds
(4/14) Running awscli tests ... done in 50 seconds
(5/14) Running healthcheck tests ... done in 0 seconds
(6/14) Running mc tests ... done in 22 seconds
(7/14) Running minio-go tests ... FAILED in 16 seconds
{
  "args": {
    "bucketName": "minio-go-test-j90vgkjj4zmmeetw",
    "objectName": "m22x0lrxcjmnck4gfd144oxileolgk"
  },
  "duration": 125,
  "function": "GetObject()",
  "message": "Unexpected error during listing objects",
  "name": "minio-go: testPutObjectWithVersioning",
  "status": "FAIL"
}
(7/14) Running minio-java tests ... FAILED in 1 minutes and 3 seconds
Exception in thread "main" java.lang.AssertionError: object count; expected=6, got=0 expected:<0> but was:<6>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at FunctionalTest.testListObjects(FunctionalTest.java:1412)
	at FunctionalTest.listObjects(FunctionalTest.java:1444)
	at FunctionalTest.runObjectTests(FunctionalTest.java:3743)
	at FunctionalTest.runTests(FunctionalTest.java:3781)
	at FunctionalTest.runEndpointTests(FunctionalTest.java:3881)
	at FunctionalTest.main(FunctionalTest.java:4002)
(7/14) Running minio-js tests ... FAILED in 20 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 5,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:71:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:166:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:90:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1333:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(7/14) Running minio-py tests ... done in 52 seconds
(8/14) Running s3cmd tests ... done in 24 seconds
(9/14) Running s3select tests ... done in 2 seconds
(10/14) Running versioning tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-2up1hqt3leca1u",
    "expiry": 60000000000,
    "objectName": "testObject"
  },
  "duration": 33,
  "error": "unexpected number of versions",
  "function": "/mint/build/versioning/put.go:108: main.testPutObject",
  "message": "Unexpected list content",
  "name": "versioning",
  "status": "FAIL"
}

Executed 9 out of 14 tests successfully.
```
<a name="e9532322-c686-4c74-a2a3-a55adad377b1"></a>
### 15511-dd6105f/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s7:32395
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 8 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-x5qy6qm2cj6pyh",
    "objectName": "4bzu6via0w30saow2q96y2f0kwnhlh"
  },
  "duration": 68,
  "function": "testObjectTagging",
  "message": "AWS SDK Go PUTObject Tag input did not match with GetObjectTagging output <nil>",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-java tests ... done in 2 seconds
(2/14) Running aws-sdk-php tests ... done in 50 seconds
(3/14) Running aws-sdk-ruby tests ... done in 5 seconds
(4/14) Running awscli tests ... done in 52 seconds
(5/14) Running healthcheck tests ... done in 0 seconds
(6/14) Running mc tests ... done in 29 seconds
(7/14) Running minio-go tests ... FAILED in 28 seconds
{
  "args": {
    "bucketName": "minio-go-test-jrgnzpd5huoze5nz",
    "objectName": "aj45q42otx6fnwxxv3ela26xecmpof"
  },
  "duration": 1129,
  "function": "GetObject()",
  "message": "Unexpected error during listing objects",
  "name": "minio-go: testPutObjectWithVersioning",
  "status": "FAIL"
}
(7/14) Running minio-java tests ... FAILED in 1 minutes and 7 seconds
Exception in thread "main" java.lang.AssertionError: object count; expected=6, got=0 expected:<0> but was:<6>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at FunctionalTest.testListObjects(FunctionalTest.java:1412)
	at FunctionalTest.listObjects(FunctionalTest.java:1444)
	at FunctionalTest.runObjectTests(FunctionalTest.java:3743)
	at FunctionalTest.runTests(FunctionalTest.java:3781)
	at FunctionalTest.runEndpointTests(FunctionalTest.java:3881)
	at FunctionalTest.main(FunctionalTest.java:4002)
(7/14) Running minio-js tests ... FAILED in 20 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 5,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:71:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:166:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:90:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1333:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(7/14) Running minio-py tests ... done in 1 minutes and 26 seconds
(8/14) Running s3cmd tests ... done in 25 seconds
(9/14) Running s3select tests ... done in 4 seconds
(10/14) Running versioning tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-fokaih4kprf6uf",
    "expiry": 60000000000,
    "objectName": "testObject"
  },
  "duration": 157,
  "error": "unexpected number of versions",
  "function": "/mint/build/versioning/put.go:108: main.testPutObject",
  "message": "Unexpected list content",
  "name": "versioning",
  "status": "FAIL"
}

Executed 9 out of 14 tests successfully.
```
<a name="50c6bebf-1a6c-4806-822f-eefe779452b9"></a>
### 15511-dd6105f/mint-gateway-s3.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s7:30347
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 38 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-0gqzt14zvlsgi5",
    "objectName": "6gtrukksczbjx90j3fe0qjcsz6cm66"
  },
  "duration": 20,
  "function": "testObjectTagging",
  "message": "AWS SDK Go PUTObject Tag input did not match with GetObjectTagging output <nil>",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-java tests ... done in 1 seconds
(2/14) Running aws-sdk-php tests ... done in 43 seconds
(3/14) Running aws-sdk-ruby tests ... done in 4 seconds
(4/14) Running awscli tests ... done in 57 seconds
(5/14) Running healthcheck tests ... done in 0 seconds
(6/14) Running mc tests ... done in 23 seconds
(7/14) Running minio-go tests ... FAILED in 36 seconds
{
  "args": {
    "bucketName": "minio-go-test-yodpwcivzfdzt36a",
    "objectPrefix": "",
    "recursive": "true"
  },
  "duration": 60,
  "function": "ListObjects(bucketName, objectPrefix, recursive, doneCh)",
  "message": "ListObjects failed unexpectedly",
  "name": "minio-go: testListObjects",
  "status": "FAIL"
}
(7/14) Running minio-java tests ... FAILED in 1 minutes and 32 seconds
Exception in thread "main" java.lang.AssertionError: tags: expected: {Project=Project One, User=jsmith}, got: {} expected:<{Project=Project One, User=jsmith}> but was:<{}>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at FunctionalTest.getObjectTags(FunctionalTest.java:3453)
	at FunctionalTest.runObjectTests(FunctionalTest.java:3771)
	at FunctionalTest.runTests(FunctionalTest.java:3781)
	at FunctionalTest.runEndpointTests(FunctionalTest.java:3881)
	at FunctionalTest.main(FunctionalTest.java:4002)
(7/14) Running minio-js tests ... FAILED in 15 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 6,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:71:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:166:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:90:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1333:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(7/14) Running minio-py tests ... done in 55 seconds
(8/14) Running s3cmd tests ... done in 1 minutes and 6 seconds
(9/14) Running s3select tests ... done in 2 seconds
(10/14) Running versioning tests ... done in 1 seconds

Executed 10 out of 14 tests successfully.
```
<a name="73adc3b3-b42d-4542-964b-f68f94e94955"></a>
### 15511-dd6105f/mint-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s1:30624
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 8 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-zytsbdfcf2c4vr",
    "objectName": "z3zllhtl3citewh6pi9c6xdjq65mgz"
  },
  "duration": 50,
  "function": "testObjectTagging",
  "message": "AWS SDK Go PUTObject Tag input did not match with GetObjectTagging output <nil>",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-java tests ... done in 2 seconds
(2/14) Running aws-sdk-php tests ... done in 43 seconds
(3/14) Running aws-sdk-ruby tests ... done in 5 seconds
(4/14) Running awscli tests ... done in 55 seconds
(5/14) Running healthcheck tests ... done in 0 seconds
(6/14) Running mc tests ... done in 24 seconds
(7/14) Running minio-go tests ... FAILED in 16 seconds
{
  "args": {
    "bucketName": "minio-go-test-vsuiioa3rjxsojeo",
    "objectName": "ptktcy6npo5tkfa4vlgoc1grx1q632"
  },
  "duration": 1099,
  "function": "GetObject()",
  "message": "Unexpected error during listing objects",
  "name": "minio-go: testPutObjectWithVersioning",
  "status": "FAIL"
}
(7/14) Running minio-java tests ... FAILED in 1 minutes and 7 seconds
Exception in thread "main" java.lang.AssertionError: object count; expected=6, got=0 expected:<0> but was:<6>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at FunctionalTest.testListObjects(FunctionalTest.java:1412)
	at FunctionalTest.listObjects(FunctionalTest.java:1444)
	at FunctionalTest.runObjectTests(FunctionalTest.java:3743)
	at FunctionalTest.runTests(FunctionalTest.java:3781)
	at FunctionalTest.runEndpointTests(FunctionalTest.java:3881)
	at FunctionalTest.main(FunctionalTest.java:4002)
(7/14) Running minio-js tests ... FAILED in 19 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 5,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:71:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:166:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:90:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1333:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(7/14) Running minio-py tests ... done in 1 minutes and 20 seconds
(8/14) Running s3cmd tests ... done in 25 seconds
(9/14) Running s3select tests ... done in 3 seconds
(10/14) Running versioning tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-0adxxn42rltqz5",
    "expiry": 60000000000,
    "objectName": "testObject"
  },
  "duration": 146,
  "error": "unexpected number of versions",
  "function": "/mint/build/versioning/put.go:108: main.testPutObject",
  "message": "Unexpected list content",
  "name": "versioning",
  "status": "FAIL"
}

Executed 9 out of 14 tests successfully.
```
<a name="909a763d-accd-45a3-8233-d225a21dcc55"></a>
### 15511-dd6105f/mint-pools.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s5:31774
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-i3y2tgn0eawerv",
    "objectName": "heiwt6q6ilezzeczf360muaawqv95j"
  },
  "duration": 75,
  "function": "testObjectTagging",
  "message": "AWS SDK Go PUTObject Tag input did not match with GetObjectTagging output <nil>",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-java tests ... done in 1 seconds
(2/14) Running aws-sdk-php tests ... done in 45 seconds
(3/14) Running aws-sdk-ruby tests ... done in 5 seconds
(4/14) Running awscli tests ... done in 56 seconds
(5/14) Running healthcheck tests ... done in 0 seconds
(6/14) Running mc tests ... done in 26 seconds
(7/14) Running minio-go tests ... FAILED in 21 seconds
{
  "args": {
    "bucketName": "minio-go-test-xtgmeszxulh3q214",
    "objectName": "6phf5k24c1ob9q6xw0ea5lru1jajiq"
  },
  "duration": 1009,
  "function": "GetObject()",
  "message": "Unexpected error during listing objects",
  "name": "minio-go: testPutObjectWithVersioning",
  "status": "FAIL"
}
(7/14) Running minio-java tests ... FAILED in 1 minutes and 8 seconds
Exception in thread "main" java.lang.AssertionError: object count; expected=6, got=0 expected:<0> but was:<6>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at FunctionalTest.testListObjects(FunctionalTest.java:1412)
	at FunctionalTest.listObjects(FunctionalTest.java:1444)
	at FunctionalTest.runObjectTests(FunctionalTest.java:3743)
	at FunctionalTest.runTests(FunctionalTest.java:3781)
	at FunctionalTest.runEndpointTests(FunctionalTest.java:3881)
	at FunctionalTest.main(FunctionalTest.java:4002)
(7/14) Running minio-js tests ... FAILED in 18 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 6,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:71:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:166:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:90:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1333:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(7/14) Running minio-py tests ... done in 1 minutes and 35 seconds
(8/14) Running s3cmd tests ... done in 26 seconds
(9/14) Running s3select tests ... done in 4 seconds
(10/14) Running versioning tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-bofmlhtmm2cofi",
    "expiry": 60000000000,
    "objectName": "testObject"
  },
  "duration": 165,
  "error": "unexpected number of versions",
  "function": "/mint/build/versioning/put.go:108: main.testPutObject",
  "message": "Unexpected list content",
  "name": "versioning",
  "status": "FAIL"
}

Executed 9 out of 14 tests successfully.
```
<a name="422f58bf-b33b-434d-93dc-1e5b73dee1a6"></a>
### 15511-dd6105f/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s2:32092
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-d6goesw1tslbyv",
    "objectName": "3urr49oez5oxgb5xc2haou1rnet1pu"
  },
  "duration": 58,
  "function": "testObjectTagging",
  "message": "AWS SDK Go PUTObject Tag input did not match with GetObjectTagging output <nil>",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-java tests ... done in 1 seconds
(2/14) Running aws-sdk-php tests ... done in 2 minutes and 39 seconds
(3/14) Running aws-sdk-ruby tests ... done in 4 seconds
(4/14) Running awscli tests ... done in 56 seconds
(5/14) Running healthcheck tests ... done in 0 seconds
(6/14) Running mc tests ... done in 29 seconds
(7/14) Running minio-go tests ... FAILED in 19 seconds
{
  "args": {
    "bucketName": "minio-go-test-6njd4ml29pikt5we",
    "objectName": "v3xsjwbocianwbqudoe4letj1yt0kz"
  },
  "duration": 893,
  "function": "GetObject()",
  "message": "Unexpected error during listing objects",
  "name": "minio-go: testPutObjectWithVersioning",
  "status": "FAIL"
}
(7/14) Running minio-java tests ... FAILED in 1 minutes and 7 seconds
Exception in thread "main" java.lang.AssertionError: object count; expected=6, got=0 expected:<0> but was:<6>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at FunctionalTest.testListObjects(FunctionalTest.java:1412)
	at FunctionalTest.listObjects(FunctionalTest.java:1444)
	at FunctionalTest.runObjectTests(FunctionalTest.java:3743)
	at FunctionalTest.runTests(FunctionalTest.java:3781)
	at FunctionalTest.runEndpointTests(FunctionalTest.java:3881)
	at FunctionalTest.main(FunctionalTest.java:4002)
(7/14) Running minio-js tests ... FAILED in 19 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 6,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:71:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:166:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:90:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1333:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(7/14) Running minio-py tests ... done in 1 minutes and 18 seconds
(8/14) Running s3cmd tests ... done in 25 seconds
(9/14) Running s3select tests ... done in 4 seconds
(10/14) Running versioning tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-heqmfo6dlpujfp",
    "expiry": 60000000000,
    "objectName": "testObject"
  },
  "duration": 140,
  "error": "unexpected number of versions",
  "function": "/mint/build/versioning/put.go:108: main.testPutObject",
  "message": "Unexpected list content",
  "name": "versioning",
  "status": "FAIL"
}

Executed 9 out of 14 tests successfully.
```
<a name="1ff432f6-63cc-4baa-bc30-ec10e375104e"></a>
### 15511-dd6105f/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s5:30053
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 8 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-fc9m9ej5z0jold",
    "objectName": "3wqqm0eglifodvccjncbf211aey02w"
  },
  "duration": 9,
  "function": "testObjectTagging",
  "message": "AWS SDK Go PUTObject Tag input did not match with GetObjectTagging output <nil>",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-java tests ... done in 1 seconds
(2/14) Running aws-sdk-php tests ... done in 43 seconds
(3/14) Running aws-sdk-ruby tests ... done in 4 seconds
(4/14) Running awscli tests ... done in 51 seconds
(5/14) Running healthcheck tests ... done in 0 seconds
(6/14) Running mc tests ... done in 22 seconds
(7/14) Running minio-go tests ... FAILED in 14 seconds
{
  "args": {
    "bucketName": "minio-go-test-mehemiux3bfiq5o4",
    "objectName": "bzamx52rfokvk0vqtynef6emiovw0c"
  },
  "duration": 107,
  "function": "GetObject()",
  "message": "Unexpected error during listing objects",
  "name": "minio-go: testPutObjectWithVersioning",
  "status": "FAIL"
}
(7/14) Running minio-java tests ... FAILED in 1 minutes and 3 seconds
Exception in thread "main" java.lang.AssertionError: object count; expected=6, got=0 expected:<0> but was:<6>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at FunctionalTest.testListObjects(FunctionalTest.java:1412)
	at FunctionalTest.listObjects(FunctionalTest.java:1444)
	at FunctionalTest.runObjectTests(FunctionalTest.java:3743)
	at FunctionalTest.runTests(FunctionalTest.java:3781)
	at FunctionalTest.runEndpointTests(FunctionalTest.java:3881)
	at FunctionalTest.main(FunctionalTest.java:4002)
(7/14) Running minio-js tests ... FAILED in 18 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 8,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:71:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:166:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:90:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1333:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(7/14) Running minio-py tests ... done in 47 seconds
(8/14) Running s3cmd tests ... done in 22 seconds
(9/14) Running s3select tests ... done in 2 seconds
(10/14) Running versioning tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-5gz9b1mu4ixybr",
    "expiry": 60000000000,
    "objectName": "testObject"
  },
  "duration": 31,
  "error": "unexpected number of versions",
  "function": "/mint/build/versioning/put.go:108: main.testPutObject",
  "message": "Unexpected list content",
  "name": "versioning",
  "status": "FAIL"
}

Executed 9 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/buildscripts/unaligned-healing.sh
+++ b/buildscripts/unaligned-healing.sh
@@ -120,7 +120,7 @@ function start_minio_16drive() {
 	cat "${WORK_DIR}/server1.log"
 	echo "FAILED"
 	mkdir -p inspects
-	(cd inspects; "${WORK_DIR}/mc" admin inspect minio/healing-shard-bucket/unaligned/**)
+	(cd inspects; "${WORK_DIR}/mc" support inspect minio/healing-shard-bucket/unaligned/**)
 
 	"${WORK_DIR}/mc" mb play/inspects
 	"${WORK_DIR}/mc" mirror inspects play/inspects
@@ -140,7 +140,7 @@ function start_minio_16drive() {
 	cat "${WORK_DIR}/server1.log"
 	echo "FAILED"
 	mkdir -p inspects
-	(cd inspects; "${WORK_DIR}/mc" admin inspect minio/healing-shard-bucket/unaligned/**)
+	(cd inspects; "${WORK_DIR}/mc" support inspect minio/healing-shard-bucket/unaligned/**)
 
 	"${WORK_DIR}/mc" mb play/inspects
 	"${WORK_DIR}/mc" mirror inspects play/inspects

--- a/cmd/api-headers.go
+++ b/cmd/api-headers.go
@@ -20,7 +20,6 @@ package cmd
 import (
 	"bytes"
 	"encoding/json"
-	"encoding/xml"
 	"fmt"
 	"net/http"
 	"net/url"
@@ -30,6 +29,8 @@ import (
 
 	"github.com/minio/minio/internal/crypto"
 	xhttp "github.com/minio/minio/internal/http"
+	"github.com/minio/minio/internal/logger"
+	xxml "github.com/minio/xxml"
 )
 
 // Returns a hexadecimal representation of time at the
@@ -64,9 +65,13 @@ func setCommonHeaders(w http.ResponseWriter) {
 // Encodes the response headers into XML format.
 func encodeResponse(response interface{}) []byte {
 	var bytesBuffer bytes.Buffer
-	bytesBuffer.WriteString(xml.Header)
-	e := xml.NewEncoder(&bytesBuffer)
-	e.Encode(response)
+	bytesBuffer.WriteString(xxml.Header)
+	buf, err := xxml.Marshal(response)
+	if err != nil {
+		logger.LogIf(GlobalContext, err)
+		return nil
+	}
+	bytesBuffer.Write(buf)
 	return bytesBuffer.Bytes()
 }
 

--- a/cmd/api-response.go
+++ b/cmd/api-response.go
@@ -89,7 +89,8 @@ type ListVersionsResponse struct {
 	IsTruncated bool
 
 	CommonPrefixes []CommonPrefix
-	Versions       []ObjectVersion
+	DeleteMarkers  []DeleteMarkerVersion `xml:"DeleteMarker,omitempty"`
+	Versions       []ObjectVersion       `xml:"Version,omitempty"`
 
 	// Encoding type used to encode object keys in the response.
 	EncodingType string `xml:"EncodingType,omitempty"`
@@ -247,50 +248,80 @@ type ObjectVersion struct {
 	Object
 	IsLatest  bool
 	VersionID string `xml:"VersionId"`
-
-	isDeleteMarker bool
 }
 
-// MarshalXML - marshal ObjectVersion
-func (o ObjectVersion) MarshalXML(e *xml.Encoder, start xml.StartElement) error {
-	if o.isDeleteMarker {
-		start.Name.Local = "DeleteMarker"
-	} else {
-		start.Name.Local = "Version"
+// DeleteMarkerVersion container for delete marker metadata
+type DeleteMarkerVersion struct {
+	Key          string
+	LastModified string // time string of format "2006-01-02T15:04:05.000Z"
+
+	// Owner of the object.
+	Owner Owner
+
+	IsLatest  bool
+	VersionID string `xml:"VersionId"`
+}
+
+// Metadata metadata items implemented to ensure XML marshaling works.
+type Metadata struct {
+	Items []struct {
+		Key   string
+		Value string
 	}
-
-	type objectVersionWrapper ObjectVersion
-	return e.EncodeElement(objectVersionWrapper(o), start)
 }
 
-// StringMap is a map[string]string
-type StringMap map[string]string
+// Set add items, duplicate items get replaced.
+func (s *Metadata) Set(k, v string) {
+	for i, item := range s.Items {
+		if item.Key == k {
+			s.Items[i] = struct {
+				Key   string
+				Value string
+			}{
+				Key:   k,
+				Value: v,
+			}
+			return
+		}
+	}
+	s.Items = append(s.Items, struct {
+		Key   string
+		Value string
+	}{
+		Key:   k,
+		Value: v,
+	})
+}
+
+type xmlKeyEntry struct {
+	XMLName xml.Name
+	Value   string `xml:",chardata"`
+}
 
 // MarshalXML - StringMap marshals into XML.
-func (s StringMap) MarshalXML(e *xml.Encoder, start xml.StartElement) error {
-	tokens := []xml.Token{start}
-
-	for key, value := range s {
-		t := xml.StartElement{}
-		t.Name = xml.Name{
-			Space: "",
-			Local: key,
-		}
-		tokens = append(tokens, t, xml.CharData(value), xml.EndElement{Name: t.Name})
+func (s *Metadata) MarshalXML(e *xml.Encoder, start xml.StartElement) error {
+	if s == nil {
+		return nil
 	}
 
-	tokens = append(tokens, xml.EndElement{
-		Name: start.Name,
-	})
+	if len(s.Items) == 0 {
+		return nil
+	}
 
-	for _, t := range tokens {
-		if err := e.EncodeToken(t); err != nil {
+	if err := e.EncodeToken(start); err != nil {
+		return err
+	}
+
+	for _, item := range s.Items {
+		if err := e.Encode(xmlKeyEntry{
+			XMLName: xml.Name{Local: item.Key},
+			Value:   item.Value,
+		}); err != nil {
 			return err
 		}
 	}
 
-	// flush to ensure tokens are written
-	return e.Flush()
+	return e.EncodeToken(start.End())
 }
 
 // Object container for object metadata
@@ -307,7 +338,7 @@ type Object struct {
 	StorageClass string
 
 	// UserMetadata user-defined metadata
-	UserMetadata StringMap `xml:"UserMetadata,omitempty"`
+	UserMetadata *Metadata `xml:"UserMetadata,omitempty"`
 }
 
 // CopyObjectResponse container returns ETag and LastModified of the successfully copied object
@@ -438,6 +469,8 @@ func generateListBucketsResponse(buckets []BucketInfo) ListBucketsResponse {
 // generates an ListBucketVersions response for the said bucket with other enumerated options.
 func generateListVersionsResponse(bucket, prefix, marker, versionIDMarker, delimiter, encodingType string, maxKeys int, resp ListObjectVersionsInfo) ListVersionsResponse {
 	versions := make([]ObjectVersion, 0, len(resp.Objects))
+	deleteMarkers := make([]DeleteMarkerVersion, 0, len(resp.Objects))
+
 	owner := Owner{
 		ID:          globalMinioDefaultOwnerID,
 		DisplayName: "minio",
@@ -445,10 +478,25 @@ func generateListVersionsResponse(bucket, prefix, marker, versionIDMarker, delim
 	data := ListVersionsResponse{}
 
 	for _, object := range resp.Objects {
-		content := ObjectVersion{}
 		if object.Name == "" {
 			continue
 		}
+
+		if object.DeleteMarker {
+			deleteMarker := DeleteMarkerVersion{
+				Key:          s3EncodeName(object.Name, encodingType),
+				LastModified: object.ModTime.UTC().Format(iso8601TimeFormat),
+				Owner:        owner,
+				VersionID:    object.VersionID,
+			}
+			if deleteMarker.VersionID == "" {
+				deleteMarker.VersionID = nullVersionID
+			}
+			deleteMarker.IsLatest = object.IsLatest
+			deleteMarkers = append(deleteMarkers, deleteMarker)
+			continue
+		}
+		content := ObjectVersion{}
 		content.Key = s3EncodeName(object.Name, encodingType)
 		content.LastModified = object.ModTime.UTC().Format(iso8601TimeFormat)
 		if object.ETag != "" {
@@ -466,12 +514,12 @@ func generateListVersionsResponse(bucket, prefix, marker, versionIDMarker, delim
 			content.VersionID = nullVersionID
 		}
 		content.IsLatest = object.IsLatest
-		content.isDeleteMarker = object.DeleteMarker
 		versions = append(versions, content)
 	}
 
 	data.Name = bucket
 	data.Versions = versions
+	data.DeleteMarkers = deleteMarkers
 	data.EncodingType = encodingType
 	data.Prefix = s3EncodeName(prefix, encodingType)
 	data.KeyMarker = s3EncodeName(marker, encodingType)
@@ -569,14 +617,14 @@ func generateListObjectsV2Response(bucket, prefix, token, nextToken, startAfter,
 		}
 		content.Owner = owner
 		if metadata {
-			content.UserMetadata = make(StringMap)
+			content.UserMetadata = &Metadata{}
 			switch kind, _ := crypto.IsEncrypted(object.UserDefined); kind {
 			case crypto.S3:
-				content.UserMetadata[xhttp.AmzServerSideEncryption] = xhttp.AmzEncryptionAES
+				content.UserMetadata.Set(xhttp.AmzServerSideEncryption, xhttp.AmzEncryptionAES)
 			case crypto.S3KMS:
-				content.UserMetadata[xhttp.AmzServerSideEncryption] = xhttp.AmzEncryptionKMS
+				content.UserMetadata.Set(xhttp.AmzServerSideEncryption, xhttp.AmzEncryptionKMS)
 			case crypto.SSEC:
-				content.UserMetadata[xhttp.AmzServerSideEncryptionCustomerAlgorithm] = xhttp.AmzEncryptionAES
+				content.UserMetadata.Set(xhttp.AmzServerSideEncryptionCustomerAlgorithm, xhttp.AmzEncryptionAES)
 			}
 			for k, v := range CleanMinioInternalMetadataKeys(object.UserDefined) {
 				if strings.HasPrefix(strings.ToLower(k), ReservedMetadataPrefixLower) {
@@ -588,7 +636,7 @@ func generateListObjectsV2Response(bucket, prefix, token, nextToken, startAfter,
 				if equals(k, xhttp.AmzMetaUnencryptedContentLength, xhttp.AmzMetaUnencryptedContentMD5) {
 					continue
 				}
-				content.UserMetadata[k] = v
+				content.UserMetadata.Set(k, v)
 			}
 		}
 		contents = append(contents, content)

--- a/cmd/object-handlers.go
+++ b/cmd/object-handlers.go
@@ -20,6 +20,7 @@ package cmd
 import (
 	"context"
 	"encoding/hex"
+	"encoding/xml"
 	"errors"
 	"fmt"
 	"io"
@@ -3066,6 +3067,16 @@ func (api objectAPIHandlers) GetObjectRetentionHandler(w http.ResponseWriter, r 
 	})
 }
 
+// ObjectTagSet key value tags
+type ObjectTagSet struct {
+	Tags []tags.Tag `xml:"Tag"`
+}
+
+type objectTagging struct {
+	XMLName xml.Name      `xml:"Tagging"`
+	TagSet  *ObjectTagSet `xml:"TagSet"`
+}
+
 // GetObjectTaggingHandler - GET object tagging
 func (api objectAPIHandlers) GetObjectTaggingHandler(w http.ResponseWriter, r *http.Request) {
 	ctx := newContext(r, w, "GetObjectTagging")
@@ -3103,7 +3114,7 @@ func (api objectAPIHandlers) GetObjectTaggingHandler(w http.ResponseWriter, r *h
 	}
 
 	// Get object tags
-	tags, err := objAPI.GetObjectTags(ctx, bucket, object, opts)
+	ot, err := objAPI.GetObjectTags(ctx, bucket, object, opts)
 	if err != nil {
 		writeErrorResponse(ctx, w, toAPIError(ctx, err), r.URL)
 		return
@@ -3113,7 +3124,20 @@ func (api objectAPIHandlers) GetObjectTaggingHandler(w http.ResponseWriter, r *h
 		w.Header()[xhttp.AmzVersionID] = []string{opts.VersionID}
 	}
 
-	writeSuccessResponseXML(w, encodeResponse(tags))
+	otags := &objectTagging{
+		TagSet: &ObjectTagSet{},
+	}
+
+	var list []tags.Tag
+	for k, v := range ot.ToMap() {
+		list = append(list, tags.Tag{
+			Key:   k,
+			Value: v,
+		})
+	}
+	otags.TagSet.Tags = list
+
+	writeSuccessResponseXML(w, encodeResponse(otags))
 }
 
 // PutObjectTaggingHandler - PUT object tagging

--- a/cmd/server_test.go
+++ b/cmd/server_test.go
@@ -28,6 +28,7 @@ import (
 	"net/http"
 	"net/url"
 	"reflect"
+	"runtime"
 	"strings"
 	"sync"
 	"testing"
@@ -1616,6 +1617,81 @@ func (s *TestSuiteCommon) TestListObjectsHandler(c *check) {
 			},
 		},
 		{getListObjectsV2URL(s.endPoint, bucketName, "", "1000", "", "url"), []string{"<Key>foo+bar+1</Key>", "<Key>foo+bar+2</Key>"}},
+	}
+
+	for _, testCase := range testCases {
+		// create listObjectsV1 request with valid parameters
+		request, err = newTestSignedRequest(http.MethodGet, testCase.getURL, 0, nil, s.accessKey, s.secretKey, s.signer)
+		c.Assert(err, nil)
+		// execute the HTTP request.
+		response, err = s.client.Do(request)
+		c.Assert(err, nil)
+		c.Assert(response.StatusCode, http.StatusOK)
+
+		getContent, err := ioutil.ReadAll(response.Body)
+		c.Assert(err, nil)
+
+		for _, expectedStr := range testCase.expectedStrings {
+			c.Assert(strings.Contains(string(getContent), expectedStr), true)
+		}
+	}
+}
+
+// TestListObjectsSpecialCharactersHandler - Setting valid parameters to List Objects
+// and then asserting the response with the expected one.
+func (s *TestSuiteCommon) TestListObjectsSpecialCharactersHandler(c *check) {
+	if runtime.GOOS == globalWindowsOSName {
+		c.Skip("skip special character test for windows")
+	}
+
+	// generate a random bucket name.
+	bucketName := getRandomBucketName()
+	// HTTP request to create the bucket.
+	request, err := newTestSignedRequest(http.MethodPut, getMakeBucketURL(s.endPoint, bucketName),
+		0, nil, s.accessKey, s.secretKey, s.signer)
+	c.Assert(err, nil)
+
+	// execute the HTTP request to create bucket.
+	response, err := s.client.Do(request)
+	c.Assert(err, nil)
+	c.Assert(response.StatusCode, http.StatusOK)
+
+	for _, objectName := range []string{"foo bar 1", "foo bar 2", "foo \x01 bar"} {
+		buffer := bytes.NewReader([]byte("Hello World"))
+		request, err = newTestSignedRequest(http.MethodPut, getPutObjectURL(s.endPoint, bucketName, objectName),
+			int64(buffer.Len()), buffer, s.accessKey, s.secretKey, s.signer)
+		c.Assert(err, nil)
+
+		response, err = s.client.Do(request)
+		c.Assert(err, nil)
+		c.Assert(response.StatusCode, http.StatusOK)
+	}
+
+	testCases := []struct {
+		getURL          string
+		expectedStrings []string
+	}{
+		{getListObjectsV1URL(s.endPoint, bucketName, "", "1000", ""), []string{"<Key>foo bar 1</Key>", "<Key>foo bar 2</Key>", "<Key>foo &#x1; bar</Key>"}},
+		{getListObjectsV1URL(s.endPoint, bucketName, "", "1000", "url"), []string{"<Key>foo+bar+1</Key>", "<Key>foo+bar+2</Key>", "<Key>foo+%01+bar</Key>"}},
+		{
+			getListObjectsV2URL(s.endPoint, bucketName, "", "1000", "", ""),
+			[]string{
+				"<Key>foo bar 1</Key>",
+				"<Key>foo bar 2</Key>",
+				"<Key>foo &#x1; bar</Key>",
+				fmt.Sprintf("<Owner><ID>%s</ID><DisplayName>minio</DisplayName></Owner>", globalMinioDefaultOwnerID),
+			},
+		},
+		{
+			getListObjectsV2URL(s.endPoint, bucketName, "", "1000", "true", ""),
+			[]string{
+				"<Key>foo bar 1</Key>",
+				"<Key>foo bar 2</Key>",
+				"<Key>foo &#x1; bar</Key>",
+				fmt.Sprintf("<Owner><ID>%s</ID><DisplayName>minio</DisplayName></Owner>", globalMinioDefaultOwnerID),
+			},
+		},
+		{getListObjectsV2URL(s.endPoint, bucketName, "", "1000", "", "url"), []string{"<Key>foo+bar+1</Key>", "<Key>foo+bar+2</Key>", "<Key>foo+%01+bar</Key>"}},
 	}
 
 	for _, testCase := range testCases {

--- a/go.mod
+++ b/go.mod
@@ -55,6 +55,7 @@ require (
 	github.com/minio/sha256-simd v1.0.0
 	github.com/minio/simdjson-go v0.4.2
 	github.com/minio/sio v0.3.0
+	github.com/minio/xxml v0.0.3
 	github.com/minio/zipindex v0.2.1
 	github.com/mitchellh/go-homedir v1.1.0
 	github.com/nats-io/nats-server/v2 v2.7.4

